Salt ctable hashing

kbara opened the pull request:
This should avoid quadratic behaviour when copying ctables.
Fixes https://github.com/Igalia/snabb/issues/654
Note that this serializes the salt; loads are a raw slurp that does not go
quadratic, and must share salt. The issue is only when copying a ctable.

wingo commented on src/lib/ctable.lua:
```diff
@@ -102,6 +102,16 @@ local function parse_params(params, required, optional)
    return ret
 end
 
+-- This gets 16 bits of salt, which should be enough to prevent accidental
```
Since our hashes are 32 bits I think probably we should just get 32 bits of salt -- otherwise I think we don't actually solve the problem.  In difference to the rust hash table, we map [0, HASH_MAX) to the [0, size) domain by scaling, not modulo, so high bits are important too.

kbara replied:
Fair point, and I'll fix that.

wingo commented on src/lib/ctable.lua:
```diff
@@ -110,15 +120,22 @@ local optional_params = {
    max_occupancy_rate = 0.9,
-   min_occupancy_rate = 0.0
+   min_occupancy_rate = 0.0,
+   salt = get_random_salt()
```
This default is set globally -- all hash tables in this process will default to this salt value.  Probably not what you want.

kbara replied:
Definitely not; good catch.

Commit:
This should avoid quadratic behaviour when copying ctables.

Fixes #654
Note that this serializes the salt; loads are a raw slurp that does not go
quadratic, and must share salt. The issue is only when copying a ctable.

kbara commented:
Hydra shows no performance difference: https://hydra.snabb.co/build/978312/download/2/report.html

Are you happy with that result @wingo , or do you need more benchmarks?

kbara commented:
https://hydra.snabb.co/build/978775/download/2/report.html shows no difference in packet loss until about 90% of line speed, between this branch and the lwaftr tip. After that, this branch is marginally worse.

wingo commented:
Closing in favor of https://github.com/snabbco/snabb/pull/1168, as we have moved to SipHash.
